test/ipu-xrt/add_256_using_dma_op_no_double_buffering/aie.mlir

```diff
@@ -50,10 +45,14 @@ module {
       aie.end
     }
 
-    %buffer_0_1 = aie.buffer(%tile_0_1) : memref<16xi32>
-    %buffer_0_1_0 = aie.buffer(%tile_0_1) : memref<16xi32>
-
     %memtile_dma_0_1 = aie.memtile_dma(%tile_0_1) {
+      %buffer_0_1 = aie.buffer(%tile_0_1) : memref<16xi32>
+      %buffer_0_1_0 = aie.buffer(%tile_0_1) : memref<16xi32>
+      %lock_0_1_0 = aie.lock(%tile_0_1, 0) {init = 1 : i32}
+      %lock_0_1_1 = aie.lock(%tile_0_1, 1) {init = 0 : i32}
+      %lock_0_1_2 = aie.lock(%tile_0_1, 2) {init = 1 : i32}
+      %lock_0_1_3 = aie.lock(%tile_0_1, 3) {init = 0 : i32}
+
```

Comment:
It's not clear to me that this is a net-win.. OK, the scoping for these ops gets a little bit tighter, but this is somewhat of an unusual case... most designs won't be able to leverage this.  At the same time, we expand the set of things that we need to verify....

Reply:
Kicking a hornett's nest: in reality these shouldn't be ops that float around _at all_ and should be closer to attributes of a tile (which itself should be a region rather than an SSA handle that also floats around).

> but this is somewhat of an unusual case

Every single memtile and shimtile doesn't "share" its buffers. Every single memtile connected only through/by/via streaming doesn't share its locks. Any core tile that uses private scratch space doesn't "share" buffers. To wit: I didn't invent this pattern, I only noticed it in the multitude of examples.

Reply:
Memtile DMAs can access neighboring memtiles.

Reply:
> Memtile DMAs can access neighboring memtiles.

Fair enough, I stand corrected, but it's still the case that this enables a simplification of a very common pattern.